### PR TITLE
Fix for #662: avoid unnecessary builds

### PR DESCRIPTION
**Pull request by smruthi2187**
Closing the other PR and opened this instead.

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/866)

### CODE REVIEW

File: fission/package.go [hunk 1]
```diff
@@ -145,8 +156,12 @@ func updatePackage(client *client.Client, pkg *crd.Package, envName, envNamespac
 	var srcArchiveMetadata, deployArchiveMetadata *fission.Archive
 	needToBuild := false
 
-	if len(envName) > 0 {
+	if len(envName) > 0 && len(srcArchiveName) > 0 {
```
Comment (life1347) [inline fission/package.go]: Remove `len(srcArchiveName) > 0` as we discussed.

File: fission/package.go [hunk 2]
```diff
@@ -117,6 +117,17 @@ func pkgUpdate(c *cli.Context) error {
 	})
 	checkErr(err, "get package")
 
+	// if the new env specified is the same as the old one, no need to update package
+	// same is true for all update parameters, but, for now, we dont check all of them - because, its ok to
+	// re-write the object with same old values, we just end up getting a new resource version for the object.
+	if len(envName) > 0 && envName == pkg.Spec.Environment.Name && envNamespace == pkg.Spec.Environment.Namespace {
```
Comment (life1347) [inline fission/package.go]: `if len(envName) > 0 && envName == pkg.Spec.Environment.Name` should be enough

File: fission/package.go (hunk 1 above)
Comment (smruthi2187) [reply fission/package.go]: makes sense. thanks for pointing this out.

File: fission/package.go (hunk 2 above)
Comment (smruthi2187) [reply fission/package.go]: oops. sure.

**Comment by life1347** [comment]
LGTM

**Pull request merged**